src/bindgen/bindings.rs

```diff
@@ -215,6 +215,27 @@ impl Bindings {
 
         self.open_namespaces(&mut out);
 
+        let noreturn_required = self.functions.iter().any(|x| x.never_return);
+        if noreturn_required {
```

Comment:
I think the way we've gone about similar C / C++ annotations is a bit different. Instead of hardcoding the macro here, we just add a configuration option so that the user of the library can use whatever macro they want / need. See for example how `must_use` works.

I think by default converting `Never` to `Void` and _then_ adding the attribute if specified in the configuration would be better.

It also avoids the walk over every function, and prevents hardcoding compiler macros here. Otherwise for users that compile with MSVC you'd need to use `__declspec(noreturn)`, etc.

Wdyt? Would such a thing work for you?